Demo on Anthos GKE for AWS

Hi,

I know the repo is GCP based but before v0.3 it worked fine on a GKE on AWS cluster. Now it fails to start many services have:-

```
2020-09-30 08:24:40 | [ERROR] | report |

***************************
APPLICATION FAILED TO START
***************************

Description:

Invalid Micrometer configuration detected:

  - stackdriver.projectId was 'null' but it is required

Action:

Update your application to correct the invalid configuration.
```

I guess this is expected ? Is there an alternative for GKE on AWS ?

Thanks,

--- a/docs/environments.md
+++ b/docs/environments.md
@@ -1,0 +1,10 @@
+# Running in Kubernetes Environments
+
+
+## GKE on GCP 
+
+Use the default instructions in the [README](/README.md). When Workload Identity is enabled, use the [Workload Identity instructions](/docs/workload-identity.md).  
+
+## AWS EKS 
+
+Use the default README instructions, but note that currently, metrics and trace export to Google Cloud Operations does not currently work. In the Deployment manifests, set `ENABLE_METRICS=false` and `ENABLE_TRACING=false` to prevent any pod crashes on startup.

--- a/src/balancereader/src/main/java/anthos/samples/bankofanthos/balancereader/BalanceReaderApplication.java
+++ b/src/balancereader/src/main/java/anthos/samples/bankofanthos/balancereader/BalanceReaderApplication.java
@@ -98,8 +98,8 @@ public class BalanceReaderApplication {
             @Override
             public String projectId() {
                 String id = MetadataConfig.getProjectId();
-                if(id == null) {
-                    id = ""; 
+                if (id == null) {
+                    id = "";
                 }
                 return id;
             }

--- a/src/balancereader/src/main/java/anthos/samples/bankofanthos/balancereader/BalanceReaderApplication.java
+++ b/src/balancereader/src/main/java/anthos/samples/bankofanthos/balancereader/BalanceReaderApplication.java
@@ -97,7 +97,11 @@ public class BalanceReaderApplication {
 
             @Override
             public String projectId() {
-                return MetadataConfig.getProjectId();
+                String id = MetadataConfig.getProjectId();
+                if(id == null) {
+                    id = ""; 
+                }
+                return id;
             }
 
             @Override

--- a/src/ledgerwriter/src/main/java/anthos/samples/bankofanthos/ledgerwriter/LedgerWriterApplication.java
+++ b/src/ledgerwriter/src/main/java/anthos/samples/bankofanthos/ledgerwriter/LedgerWriterApplication.java
@@ -103,7 +103,11 @@ public class LedgerWriterApplication {
 
             @Override
             public String projectId() {
-                return MetadataConfig.getProjectId();
+                String id = MetadataConfig.getProjectId();
+                if(id == null) {
+                    id = ""; 
+                }
+                return id;
             }
 
             @Override

--- a/src/ledgerwriter/src/main/java/anthos/samples/bankofanthos/ledgerwriter/LedgerWriterApplication.java
+++ b/src/ledgerwriter/src/main/java/anthos/samples/bankofanthos/ledgerwriter/LedgerWriterApplication.java
@@ -104,8 +104,8 @@ public class LedgerWriterApplication {
             @Override
             public String projectId() {
                 String id = MetadataConfig.getProjectId();
-                if(id == null) {
-                    id = ""; 
+                if (id == null) {
+                    id = "";
                 }
                 return id;
             }

--- a/src/transactionhistory/src/main/java/anthos/samples/bankofanthos/transactionhistory/TransactionHistoryApplication.java
+++ b/src/transactionhistory/src/main/java/anthos/samples/bankofanthos/transactionhistory/TransactionHistoryApplication.java
@@ -98,12 +98,12 @@ public class TransactionHistoryApplication {
             @Override
             public String projectId() {
                 String id = MetadataConfig.getProjectId();
-                if(id == null) {
-                    id = ""; 
+                if (id == null) {
+                    id = "";
                 }
                 return id;
             }
-            
+
             @Override
             public String get(String key) {
                 return null;

--- a/src/transactionhistory/src/main/java/anthos/samples/bankofanthos/transactionhistory/TransactionHistoryApplication.java
+++ b/src/transactionhistory/src/main/java/anthos/samples/bankofanthos/transactionhistory/TransactionHistoryApplication.java
@@ -97,9 +97,13 @@ public class TransactionHistoryApplication {
 
             @Override
             public String projectId() {
-                return MetadataConfig.getProjectId();
+                String id = MetadataConfig.getProjectId();
+                if(id == null) {
+                    id = ""; 
+                }
+                return id;
             }
-
+            
             @Override
             public String get(String key) {
                 return null;

